lib/jxl/dec_render_pipeline.h

```diff
@@ -24,28 +24,31 @@ enum class RenderPipelineChannelMode {
 
 class RenderPipelineStage {
  public:
-  // `input` points to `2*MaxPaddingY() + 1` pointers, each of which points to
-  // `3+num_non_color_channels` pointer-to-row. So, `input[MaxPaddingY()][0]` is
+  // `input` points to `2*MaxBorderY() + 1` pointers, each of which points to
+  // `3+num_non_color_channels` pointer-to-row. So, `input[MaxBorderY()][0]` is
   // the pointer to the center row of the first color channel.
-  //  `MaxPaddingY()` is the maximum value returned by `GetPaddingX()`;
+  //  `MaxBorderY()` is the maximum value returned by `GetBorderX()`;
   //  typically, this is a constant.
   // `output` points to `1<<MaxShiftY()` pointers, each of which points to
   // `3+num_non_color_channels` pointer-to-row. So, `output[0][3]` is the
   //  pointer to the top row of the first non-color channel.
-  //  `MaxShiftY()` is defined similarly to `MaxPaddingY()`.
-  //  `xsize` represents the total number of pixels to be processed in the input
-  //  row. `xpos` and `ypos` represent the position of the first pixel in the
-  //  center row in the input
+  //  `MaxShiftY()` is defined similarly to `MaxBorderY()`.
+  //  `xsize+xextra` represents the total number of pixels to be processed in
+  //  the input row, where the first pixel in the row at position
+  //  `kRenderPipelineXOffset-xextra`. `xpos` and `ypos` represent the position
+  //  of the first (non-extra, i.e. in position kRenderPipelineXOffset) pixel in
+  //  the center row in the input. `xpos` is a multiple of
+  //  `GroupBorderAssigner::kPaddingXRound`.
   virtual void ProcessRow(float* JXL_RESTRICT** input,
-                          float* JXL_RESTRICT** output, size_t xsize,
-                          size_t xpos, size_t ypos) const = 0;
+                          float* JXL_RESTRICT** output, size_t xextra,
+                          size_t xsize, size_t xpos, size_t ypos) const = 0;
   virtual ~RenderPipelineStage() {}
 
   // Amount of padding required by each channel in the various directions.
   // The value for c=0 indicates padding required for color channels, subsequent
   // values refer to padding for non-color channels, in order.
-  virtual size_t GetPaddingX(size_t c) const = 0;
-  virtual size_t GetPaddingY(size_t c) const = 0;
+  virtual size_t GetBorderX(size_t c) const = 0;
+  virtual size_t GetBorderY(size_t c) const = 0;
```

Comment:
Is different border per channel a use-case we need? I think that's simplify things a bit.

Reply:
Unfortunately yes... Extra channels will have different borders in most cases.